pkg/deploy/api/types.go

```diff
@@ -169,6 +174,19 @@ const (
 	DefaultRollingUpdatePeriodSeconds int64 = 1
 )
 
+type AfterDeploymentActions struct {
+	// TagImages instructs the deployer to tag the current image referenced under a container onto an image stream tag if the deployment succeeds.
+	TagImages []TagImageAction `json:"tagImages,omitempty" description:"a list of image stream tags that should be updated to the latest value on this deployment if the deployment succeeds"`
+}
+
+// TagImageAction is a request to tag the image in a particular container onto an ImageStreamTag on a successful deployment.
+type TagImageAction struct {
+	// ContainerName is the name of a container in the deployment config whose image value will be used as the source of the tag
+	ContainerName string
+	// ImageStreamTagName is the name of an ImageStreamTag in the current namespace that will be updated if the deployment succeeds.
+	ImageStreamTagName string
```

Comment:
why constraint it to this namespace?  why not allow the user to specify a name+namespace for the tag?

suppose I want to do all my TestDeployments in a particular namespace for quota reasons, but then tag the validated images to prod?